### PR TITLE
Adding some basic stuff to client-go example

### PR DESCRIPTION
**Pull request by arehmandev**
Hi I was just playing around with the kubernetes client-go and I thought it'd be nice to have this added to the example.

Thanks

### CODE REVIEW

**Comment by arehmandev** [comment]
Signed CLA

File: staging/src/k8s.io/client-go/examples/out-of-cluster-client-configuration/README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Make sure your `kubectl` is configured and pointed to a cluster. Run
 Run this application with:
 
     cd out-of-cluster-client-configuration
+    go get -v -u ./...
```
Comment (sttts) [inline staging/src/k8s.io/client-go/examples/out-of-cluster-client-configuration/README.md]: this is not really true if you are on a release branch. Please add a comment that you have to use a vendoring tool to get the correct dependencies.

File: staging/src/k8s.io/client-go/examples/out-of-cluster-client-configuration/main.go [hunk 2]
```diff
@@ -62,15 +62,34 @@ func main() {
...
 			fmt.Printf("Error getting pod %v\n", statusError.ErrStatus.Message)
 		} else if err != nil {
 			panic(err.Error())
 		} else {
-			fmt.Printf("Found pod\n")
+			fmt.Printf("Found pods\n")
+
+			// Example of retrieving pod names, creationtimestamps by iterating through the Items
+			for _, pod := range pods.Items {
+				fmt.Println("######################")
```
Comment (sttts) [inline staging/src/k8s.io/client-go/examples/out-of-cluster-client-configuration/main.go]: Nit: Alternative: add `- ` in front of `Pod name:` and indent containers a bit more. Would be less ugly.

**Pull request closed without merging**